src/vs/workbench/services/notification/common/notificationService.ts

```diff
@@ -17,11 +17,36 @@ export class NotificationService extends Disposable implements INotificationServ
 	declare readonly _serviceBrand: undefined;
 
 	readonly model = this._register(new NotificationsModel());
+	private readonly _onDidAddNotification = new Emitter<INotification>();
+	private readonly _onDidRemoveNotification = new Emitter<INotification>();
 
 	constructor(
 		@IStorageService private readonly storageService: IStorageService
 	) {
 		super();
+		this._register(this.model.onDidChangeNotification(e => {
```

Comment:
Again, my personal preference to put listening into a `registerListeners` method.